examples/plotting/imageanimatorwcs.py

```diff
@@ -0,0 +1,112 @@
+# ``map_sequence``.
+# We will be using a overally complex method to create a header.
+
+# This dictionary comphersion is extracting the three basic keywords we need
+# to create a astropy.wcs.WCS header: 'CTYPE','CUNIT', 'CDELT', 'CRVAL'
+# and 'CRPIX' from the meta information stored in the 'map_sequence'.
+wcs_input_dict = {f'{key}{n+1}': map_sequence.all_meta()[0].get(f'{key}{n}')
+                  for n, key in product([1, 2], ['CTYPE', 'CUNIT', 'CDELT', 'CRVAL', 'CRPIX'])}
+
+# Now we need to get the time difference between the two observations.
+t0, t1 = map(parse_time, [k['date-obs'] for k in map_sequence.all_meta()])
+time_diff = (t1 - t0).to(u.s)
+wcs_input_dict.update({'CTYPE1': 'Time', 'CUNIT1': time_diff.unit.name, 'CDELT1': time_diff.value})
+
+# Now add the correct number of axes to the WCS header
+wcs_input_dict.update({'NAXIS': 3, 'NAXIS1': 2, 'NAXIS2': 1024, 'NAXIS3': 1024})
+
+# We can now just pass this into astropy.wcs.WCS to create our WCS header.
+wcs = astropy.wcs.WCS(wcs_input_dict)
+
+# Now the resulting WCS object will look like:
+print(wcs)
+
+###############################################################################
+# Now we can create the animation.
+# As `~sunpy.visualization.animator.ImageAnimatorWCS` assumes the last two axes
+# are the two form the image, we can change this  by passing in the
+# ``image_axes`` keyword.
+
+#sequence_array = np.rollaxis(sequence_array, -1)
+wcs_anim = ImageAnimatorWCS(sequence_array, wcs=wcs, vmax=1000, image_axes=[0, 1])
+
+###############################################################################
+# You might notice that the animation could do with having the axes ticks
+# being tidied up. To do this we can call
+# `~sunpy.visualization.animator.ImageAnimatorWCS.update_axes_style`.
+
+wcs_anim = ImageAnimatorWCS(sequence_array, wcs=wcs, vmax=1000, image_axes=[0, 1])
+wcs_anim.update_axes_style((20 * u.arcsec), 't', 'r')
+# Notice the title for this one.
+# Is this the bug for https://github.com/sunpy/sunpy/pull/2894?
+
+###############################################################################
+# You might notice that the animation could do with having the axes ticks
+# being tidied up. Since we are using a WCS object, we can use
+# `~astropy.visualization.wcsaxes` to change the properties of the visualization.
+# `Documentation is here <http://docs.astropy.org/en/stable/visualization/wcsaxes/index.html>_`
+
+wcs_anim = ImageAnimatorWCS(sequence_array, wcs=wcs, vmax=1000, image_axes=[0, 1])
+
+# We need to access the underlying WCS axes object.
+# This way we change various properties of our visualization.
+ax = wcs_anim.axes
+
+# Now we can label the X and Y axes.
+ax.set_xlabel('Solar X (arsec)')
+ax.set_ylabel('Solar Y (arsec)')
+
+# To change the spacing of the ticks for each axis we have to go deeper.
+# ax.coords holds each axis in a list and we can index.
+y_axis, x_axis = ax.coords[1], ax.coords[2]
```

Comment:
I suggest you use names that represent the coordinates here, i.e. `lat, lon` (not actually those) rather than `y_axis, x_axis`